### PR TITLE
Replace deprecated onBackPressed

### LINKED ISSUE #2961: replace deprecated onBackPressed
onBackPressed is deprecated with api 33

migration guides:
https://medium.com/tech-takeaways/how-to-migrate-the-deprecated-onbackpressed-function-e66bb29fa2fd
https://www.youtube.com/watch?v=UsPJLG6ztBg

### PR DESCRIPTION
**Pull request by ezhil56x**
Resolves #2961

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not needed
- [x] 🔖 Capability is checked or not needed 
- [x] 🔙 Backport requests are created or not needed: `/backport to stable-xx.x`
- [x] 📅 Milestone is set
- [x] 🌸 PR title is meaningful (if it should be in the changelog: is it meaningful to users?)

### DCO
Signed-off-by: Ezhil Shanmugham ezhil56x.contact@gmail.com

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
Hi @ezhil56x , see code check results below at the bottom of the page. Could you look into them and get them fixed or should somebody else take over? Any help in fixing is highly appreciated of course 👍

**Comment by ezhil56x** [comment]
Hi @AndyScherzinger I am not sure how can I fix these. Can you explain me in detail how can I fix these. I am interested to learn these things

**Comment by AndyScherzinger** [comment]
Hi @ezhil56x 

yes absolutely. So we run several code checks via Github actions, namely in this case detekt and ktlint. One gives you feedback on hte code formatting the other one about bad pattern in the code itself.

So for the styling (ktlint) it reports the following:
```
talk-android/talk-android/app/src/main/java/com/nextcloud/talk/conversationlist/ConversationsListActivity.kt:1370:1 Needless blank line(s)
talk-android/talk-android/app/src/main/java/com/nextcloud/talk/conversationlist/ConversationsListActivity.kt:1387:1 Needless blank line(s)
talk-android/talk-android/app/src/main/java/com/nextcloud/talk/profile/ProfileActivity.kt:95:1 Unexpected blank line(s) before "}"
talk-android/talk-android/app/src/main/java/com/nextcloud/talk/remotefilebrowser/activities/RemoteFileBrowserActivity.kt:209:1 Unexpected blank line(s) before "}"
talk-android/talk-android/app/src/main/java/com/nextcloud/talk/shareditems/activities/SharedItemsActivity.kt:63:1 Unexpected blank line(s) before "}"
talk-android/talk-android/app/src/main/java/com/nextcloud/talk/translate/TranslateActivity.kt:67:1 Unexpected blank line(s) before "}"
```

For detekt it is a bit more complicated since we measure that the number of warnings is below a certain threshold, so I can't tell you excactly what needs to be fixed, just that the number of issue grew, see https://github.com/nextcloud/talk-android/actions/runs/5013997105/jobs/8987742960?pr=3036 when you check the console output of "Check detekt" it will show many lines printed in orange color. However sicne it also detects blank lines and formatting issues, maybe having the above fixed might be just enough, since it reports the following:
```
formatting - 35min debt
	NoBlankLineBeforeRbrace - [Unexpected blank line(s) before "}"] at talk-android/app/src/main/java/com/nextcloud/talk/conversationinfo/ConversationInfoActivity.kt:116:1
	NoBlankLineBeforeRbrace - [Unexpected blank line(s) before "}"] at talk-android/app/src/main/java/com/nextcloud/talk/translate/TranslateActivity.kt:67:1
	NoBlankLineBeforeRbrace - [Unexpected blank line(s) before "}"] attalk-android/app/src/main/java/com/nextcloud/talk/profile/ProfileActivity.kt:95:1
	NoConsecutiveBlankLines - [Needless blank line(s)] at talk-android/app/src/main/java/com/nextcloud/talk/conversationlist/ConversationsListActivity.kt:1370:1
	NoConsecutiveBlankLines - [Needless blank line(s)] at talk-android/app/src/main/java/com/nextcloud/talk/conversationlist/ConversationsListActivity.kt:1387:1
	NoBlankLineBeforeRbrace - [Unexpected blank line(s) before "}"] at talk-android/app/src/main/java/com/nextcloud/talk/shareditems/activities/SharedItemsActivity.kt:63:1
	NoConsecutiveBlankLines - [Needless blank line(s)] at talk-android/app/src/main/java/com/nextcloud/talk/remotefilebrowser/activities/RemoteFileBrowserActivity.kt:209:1
```

**Comment by AndyScherzinger** [comment]
For the DCO check, the easiest way for you is to put an adapted line into the PR description here in th epattern of:

Signed-off-by: Your name your.name@address.com

---
It is the DCO -> check [Developer Certificate of Origin](https://developercertificate.org/) for more information if interested in this topic.

File: app/src/main/java/com/nextcloud/talk/profile/ProfileActivity.kt [hunk 1]
```diff
@@ -4,9 +4,11 @@
  * @author Tobias Kaminsky
  * @author Andy Scherzinger
  * @author Tim Krüger
+ * @author Ezhil Shanmugham
```
Comment (AndyScherzinger) [inline app/src/main/java/com/nextcloud/talk/profile/ProfileActivity.kt:7]: no changes in this file, so no authorship, probably a copy&paste thing

File: app/src/main/java/com/nextcloud/talk/messagesearch/MessageSearchActivity.kt [hunk 2]
```diff
@@ -2,8 +2,10 @@
  * Nextcloud Talk application
  *
  * @author Álvaro Brey
+ * @author Ezhil Shanmugham
```
Comment (AndyScherzinger) [inline app/src/main/java/com/nextcloud/talk/messagesearch/MessageSearchActivity.kt:5]: no changes in this file, so no authorship, probably a copy&paste thing

File: app/src/main/java/com/nextcloud/talk/location/LocationPickerActivity.kt [hunk 3]
```diff
@@ -2,7 +2,9 @@
  * Nextcloud Talk application
  *
  * @author Marcel Hibbe
+ * @author Ezhil Shanmugham
```
Comment (AndyScherzinger) [inline app/src/main/java/com/nextcloud/talk/location/LocationPickerActivity.kt:5]: no changes in this file, so no authorship, probably a copy&paste thing

Comment (AndyScherzinger) [reply app/src/main/java/com/nextcloud/talk/location/LocationPickerActivity.kt:5]: ...or it is lacking the back-pressed changes

File: app/src/main/java/com/nextcloud/talk/messagesearch/MessageSearchActivity.kt (hunk 2 above)
Comment (AndyScherzinger) [reply app/src/main/java/com/nextcloud/talk/messagesearch/MessageSearchActivity.kt:5]: ...or it is lacking the back-pressed changes

File: app/build.gradle [hunk 4]
```diff
@@ -301,7 +301,7 @@ dependencies {
     gplayImplementation 'com.google.android.gms:play-services-base:18.2.0'
     gplayImplementation "com.google.firebase:firebase-messaging:23.1.2"
 
-    // implementation 'androidx.activity:activity-ktx:1.4.0'
+     implementation 'androidx.activity:activity-ktx:1.6.0-rc02'
```
[the remark this answers is not in the document]
Comment (ezhil56x) [reply app/build.gradle]: I have migrated to androidx.activity:activity-ktx:1.7.1 which is the latest stable version. I went through some documentation and most of them suggested to use latest stable version as they offer bug fixes, improvements, and new features.

File: app/src/main/java/com/nextcloud/talk/messagesearch/MessageSearchActivity.kt (hunk 2 above)
Comment (ezhil56x) [reply app/src/main/java/com/nextcloud/talk/messagesearch/MessageSearchActivity.kt]: I mistakenly added authorship while adding for the rest

File: app/src/main/java/com/nextcloud/talk/location/LocationPickerActivity.kt (hunk 3 above)
Comment (ezhil56x) [reply app/src/main/java/com/nextcloud/talk/location/LocationPickerActivity.kt]: I mistakenly added authorship while adding for the rest

**Comment by ezhil56x** [comment]
> Hi @ezhil56x
> 
> yes absolutely. So we run several code checks via Github actions, namely in this case detekt and ktlint. One gives you feedback on hte code formatting the other one about bad pattern in the code itself.
> 
> So for the styling (ktlint) it reports the following:
> 
> ```
> talk-android/talk-android/app/src/main/java/com/nextcloud/talk/conversationlist/ConversationsListActivity.kt:1370:1 Needless blank line(s)
> talk-android/talk-android/app/src/main/java/com/nextcloud/talk/conversationlist/ConversationsListActivity.kt:1387:1 Needless blank line(s)
> talk-android/talk-android/app/src/main/java/com/nextcloud/talk/profile/ProfileActivity.kt:95:1 Unexpected blank line(s) before "}"
> talk-android/talk-android/app/src/main/java/com/nextcloud/talk/remotefilebrowser/activities/RemoteFileBrowserActivity.kt:209:1 Unexpected blank line(s) before "}"
> talk-android/talk-android/app/src/main/java/com/nextcloud/talk/shareditems/activities/SharedItemsActivity.kt:63:1 Unexpected blank line(s) before "}"
> talk-android/talk-android/app/src/main/java/com/nextcloud/talk/translate/TranslateActivity.kt:67:1 Unexpected blank line(s) before "}"
> ```
> 
> For detekt it is a bit more complicated since we measure that the number of warnings is below a certain threshold, so I can't tell you excactly what needs to be fixed, just that the number of issue grew, see https://github.com/nextcloud/talk-android/actions/runs/5013997105/jobs/8987742960?pr=3036 when you check the console output of "Check detekt" it will show many lines printed in orange color. However sicne it also detects blank lines and formatting issues, maybe having the above fixed might be just enough, since it reports the following:
> 
> ```
> formatting - 35min debt
> 	NoBlankLineBeforeRbrace - [Unexpected blank line(s) before "}"] at talk-android/app/src/main/java/com/nextcloud/talk/conversationinfo/ConversationInfoActivity.kt:116:1
> 	NoBlankLineBeforeRbrace - [Unexpected blank line(s) before "}"] at talk-android/app/src/main/java/com/nextcloud/talk/translate/TranslateActivity.kt:67:1
> 	NoBlankLineBeforeRbrace - [Unexpected blank line(s) before "}"] attalk-android/app/src/main/java/com/nextcloud/talk/profile/ProfileActivity.kt:95:1
> 	NoConsecutiveBlankLines - [Needless blank line(s)] at talk-android/app/src/main/java/com/nextcloud/talk/conversationlist/ConversationsListActivity.kt:1370:1
> 	NoConsecutiveBlankLines - [Needless blank line(s)] at talk-android/app/src/main/java/com/nextcloud/talk/conversationlist/ConversationsListActivity.kt:1387:1
> 	NoBlankLineBeforeRbrace - [Unexpected blank line(s) before "}"] at talk-android/app/src/main/java/com/nextcloud/talk/shareditems/activities/SharedItemsActivity.kt:63:1
> 	NoConsecutiveBlankLines - [Needless blank line(s)] at talk-android/app/src/main/java/com/nextcloud/talk/remotefilebrowser/activities/RemoteFileBrowserActivity.kt:209:1
> ```

I have made these suggested changes and now its is working fine

**Comment by ezhil56x** [comment]
> For the DCO check, the easiest way for you is to put an adapted line into the PR description here in th epattern of:
> 
> Signed-off-by: Your name [your.name@address.com](mailto:your.name@address.com)
> 
> It is the DCO -> check [Developer Certificate of Origin](https://developercertificate.org/) for more information if interested in this topic.

As you said I have added Signed-off-by: Ezhil Shanmugham ezhil56x.contact@gmail.com in the description of this PR as well as tried to include this is a commit message but still I am not able to fix this

image

**Comment by AndyScherzinger** [comment]
@ezhil56x all good! The DCO you can't fix since it'll always check the commits only. The Codacy is also fine. It detected an existing TODO comment in the code not produces by you, so the failed check can be seen as "expected".

Fingers crossed @mahibi can do a review and then we would e good to go from my point of view.

**Comment by ezhil56x** [comment]
@AndyScherzinger Thank you for your guidance on how to approach and fix these bugs 👍🏻

**Comment by mahibi** [comment]
Thank you @ezhil56x for your work :heart: and @AndyScherzinger to take over guidance and sorry i didn't find time earlier!

There are some things to fix here code wise. I will comment in the code itself..

If you like and find the time, feel free to fix yourself @ezhil56x , otherwise just let me know if i should take over.

File: app/src/main/java/com/nextcloud/talk/activities/FullScreenImageActivity.kt [hunk 5]
```diff
@@ -53,7 +61,8 @@ class FullScreenImageActivity : AppCompatActivity() {
 
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         return if (item.itemId == android.R.id.home) {
-            onBackPressed()
+            // onBackPressed()
+            onBackPressedDispatcher.addCallback(this, callback)
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/activities/FullScreenImageActivity.kt:65]: Currently this back button behaviour won't work. 


the 
`onBackPressedDispatcher.addCallback(this, callback)`
should be set in `onCreate`.

Here you just have to call onBackPressedDispatcher.onBackPressed():

```
  override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return if (item.itemId == android.R.id.home) {
               onBackPressedDispatcher.onBackPressed()
```

Same for other activities where the home button is used in onOptionsItemSelected.

**Comment by mahibi** [comment]
> > Thank you @ezhil56x for your work heart and @AndyScherzinger to take over guidance and sorry i didn't find time earlier!
> > There are some things to fix here code wise. I will comment in the code itself..
> > If you like and find the time, feel free to fix yourself @ezhil56x , otherwise just let me know if i should take over.
> 
> Hi @mahibi, Happy to hear star2 I have made the suggested changes. If there are still bugs or changes to be done can you take it from here

thank you @ezhil56x , i will have a closer look tomorrow

**Review by AndyScherzinger — APPROVED** [review]

**Pull request merged**